lib/newsFields.js

```diff
@@ -0,0 +1,25 @@
+import {
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+  GraphQLList
+} from 'graphql';
+
+import { EventType } from './eventFields.js';
+import { EventFetch, AllEventsFetch } from './eventFetch.js';
+
+export const NewsSchemaFields = {
+  news: {
+    type: EventType,
+    description: 'Request a single Red Badger news article by its ID',
+    args: {
+      id: { type: GraphQLString },
+    },
+    resolve: EventFetch.bind({docType: 'news'}),
```

Comment:
I think it would be clearer to have separate handlers for news and just pull any shared logic out for mapping/sanitising, etc, instead of binding a new context to the event handlers.